andela/mazus-ah-backend
# #167165067 User Should Be Able To Follow, Unfollow, View Followers And Following.

#What does this PR do?
Adds follow functionality to users

#### Description of Task to be completed?
Adds user can follow other users route
Adds user can unfollow other users route
Adds user can get a users followers routes
Adds user can get a users followings routes
Refactor model files to account for UUIDs change
Adds documentation for the 4 routes
Adds integration tests for the 4 routes
Adds input validator for specifically checking id params

#### How should this be manually tested?
> 1. Fetch this branch on the terminal using 
> `git fetch origin ft-user-should-follow-167165067`
> 2. Install dependencies using `npm install`
> 3. Start the server with `npm run server`
> 4. On Postman, sign up by making a POST request to /api/v1/auth/signup twice. 
> 5 Using the first account's authorization token
> 6. On Postman, make a POST request to /api/v1/profiles/follow/{second user id} to follow user.
> 7. On Postman, make a DELETE request to /api/v1/profiles/follow/{second user id} to unfollow user.
> 8. On Postman, make a GET request to /api/v1/profiles/followers/{second user id} to view users followers.
> 9. On Postman, make a GET request to /api/v1/profiles/followings/{second user id} to view users followings.
> 

#### Any background context you want to provide?
Only authenticated users should be able to follow and or unfollow other users, I need a view to manage those I follow and be able to see those who follow me

#### What are the relevant pivotal tracker stories?
[#167165067](https://www.pivotaltracker.com/story/show/167165067)

#### Screenshots (if appropriate)
<img width="1105" alt="Screenshot 2019-08-02 at 9 44 13 AM" src="https://user-images.githubusercontent.com/26358047/62367538-e14d5280-b521-11e9-94f0-8036a6b165ed.png">
<img width="1048" alt="Screenshot 2019-08-02 at 9 42 48 AM" src="https://user-images.githubusercontent.com/26358047/62367564-f88c4000-b521-11e9-828a-f99579c7e6fd.png">
<img width="1115" alt="Screenshot 2019-08-02 at 9 42 38 AM" src="https://user-images.githubusercontent.com/26358047/62367579-017d1180-b522-11e9-81d0-8d6a9e393c42.png">
![Uploading Screenshot 2019-08-02 at 9.42.21 AM.png…]()
